cmd/jujud/agent/machine/manifolds.go

```diff
@@ -928,7 +936,24 @@ func IAASManifolds(config ManifoldsConfig) dependency.Manifolds {
 			Clock:                        config.Clock,
 			NewCredentialValidatorFacade: common.NewCredentialInvalidatorFacade,
 		}))),
-	})
+	}
+
+	if utilsfeatureflag.Enabled(feature.InstanceMutater) {
+		manifolds[environTrackerName] = ifNotMigrating(environ.Manifold(environ.ManifoldConfig{
+			APICallerName:  apiCallerName,
+			NewEnvironFunc: config.NewEnvironFunc,
```

Comment:
Only the controller has access to the environment normally. What is this for?

Reply:
We need access to the LXD broker, this way allows us to get that, but is why I wanted feedback, as I'm not sure it's generally the best approach.

Reply:
I think we need a call to sort this out. Not sure it'll work well in a review situation.